woocommerce/woocommerce-ios
# Order Details > Shipping Label: implement shipping label details UI

Part of #2167 

## Why

The user can now tap on the "View Shipment Details" row in a shipping label card to view the details about a shipping label.

## Changes

Ref: p91TBi-3Lo-p2

- Added two new icons from Gridicons: `.house` and `.creditCard` for the details screen
- In `OrderDetailsDataSource`, updated `shippingLabel(at:)` to be public for tap action handling in `OrderDetailsViewModel`
- In `OrderDetailsViewModel`, showed `ShippingLabelDetailsViewController` when tapping on the shipping label details row
- Created `ShippingLabelDetailsViewController` and `ShippingLabelDetailsViewModel` that displays a table view of shipping label details

## Testing

Prerequisites: the site has [WooCommerce Shipping & Tax plugin](https://docs.woocommerce.com/document/woocommerce-shipping-and-tax/) and there is an order with at least one shipping label (refunded or not).

You can set up your store to use a test credit card with instructions in p91TBi-3xD-p2.

- Go to the orders tab
- Tap on an order with at least one shipping label
- After a shipping label card is shown, tap on "View Shipment Details" --> it should navigate to a table view of 5 rows: origin address, destination address, package details (package name), shipping carrier and rates (the rate is formatted based on site settings and currency from the label), payment method (hard-coded to "Credit Card")

## Example screenshots

Notes for design reviewers - there are two places that deviate from the iOS design:
- The destination address icon uses a solid "house" icon from [Gridicons](http://automattic.github.io/gridicons/) as I couldn't find an outlined version
- For the origin/destination address content, I followed Android to show the name and multiline formatted address (the iOS design shows a comma-separated address). Using the multiline address formatted by iOS Contacts framework ensures the order follows the country convention (you can see that the origin address from the US has a different order from the destination address to Taiwan)

dark | light
-- | --
![Simulator Screen Shot - iPhone 11 - 2020-12-03 at 10 31 37](https://user-images.githubusercontent.com/1945542/100956159-d3af3f00-3552-11eb-9704-09ba42f40bb1.png) | ![Simulator Screen Shot - iPhone 11 - 2020-12-03 at 10 31 54](https://user-images.githubusercontent.com/1945542/100956163-d742c600-3552-11eb-83df-f9b71a139f6e.png)


Update release notes:
